### DISCUSSION TITLE
How to render fully-qualified image asset url from mailer layout template?

### DISCUSSION THREAD

**Discussion by krsyoung** [ElMassimo/vite_ruby#187] [Q&A] [answered]
Hello, I'm currently testing vite_ruby as a replacement for webpacker, which in general has been an absolute dream!

I have one issue that I'm not finding information on and was curious what others thought. We have a `mailer.html.erb` layout which contains an image file. Previously we had used the `asset_pack_url` so that we would get a fully qualified url to the asset and the image would render correctly in email clients.

It doesn't appear that the same exists within vite_ruby? `vite_asset_path` works as expected, but doesn't include the FQDN and thus doesn't render correctly. I tried `vite_asset_url` but it doesn't exist (was worth a shot).

Have others run into this and maybe there is a different helper method that I'm missing?

Thanks!

**Comment by ElMassimo** [comment, marked as answer]
Hi Christopher! @krsyoung 

The best approach is to [configure `asset_host`](https://guides.rubyonrails.org/action_mailer_basics.html#adding-images-in-action-mailer-views), which will be [used implicitly](https://github.com/rails/rails/blob/fb1ab3460a676ce7def0819c2e92289ef2dcbe3b/actionview/lib/action_view/helpers/asset_url_helper.rb#L279) by tag helpers in Vite Rails.

**Reply by krsyoung** [reply]
Duh, thanks! Turns out I had the perfect combination of not testing the right thing.

I needed to add a `config.asset_host` to `development.rb` and things look good (`production.rb` has the correct value for `asset_host` but was using the old `asset_pack_url` tag).

Long story short, you're awesome! 👍
